Orange/preprocess/remove.py

```diff
@@ -0,0 +1,279 @@
+
+from .preprocess import Preprocess
+from Orange.data import Domain, DiscreteVariable, Table
+from Orange.preprocess.transformation import Lookup
+
+__all__ = ["Remove"]
+
+
+class Remove(Preprocess):
+    """
+    Construct a preprocessor for removing constant features/classes
+    and unused values.
+    Given a data table, preprocessor returns a new table and a list of
+    results. In the new table, the constant features/classes and unused
+    values are removed. The list of results consists of two dictionaries.
+    The first one contains numbers of 'removed', 'reduced' and 'sorted'
+    features. The second one contains numbers of 'removed', 'reduced'
+    and 'sorted' features.
+
+    Parameters
+    ----------
+    attr_flags : int (default: 0)
+        If SortValues, values of discrete attributes are sorted.
+        If RemoveConstant, unused attributes are removed.
+        If RemoveUnusedValues, unused values are removed from descrete
+        attributes.
+        It is possible to merge operations in one by summing several types.
+
+    class_flags: int (default: 0)
+        If SortValues, values of discrete class attributes are sorted.
+        If RemoveConstant, unused class attributes are removed.
+        If RemoveUnusedValues, unused values are removed from descrete
+        class attributes.
+        It is possible to merge operations in one by summing several types.
+
+    Examples
+    --------
+    >>> from Orange.data import Table
+    >>> from Orange.preprocess import Remove
+    >>> data = Table("zoo")[:10]
+    >>> flags = sum([Remove.SortValues, Remove.RemoveConstant, Remove.RemoveUnusedValues])
+    >>> remover = Remove(attr_flags=flags, class_flags=flags)
+    >>> new_data, [attr_res, class_res] = remover(data)
+    """
+
+    SortValues, RemoveConstant, RemoveUnusedValues = 1, 2, 4
+
+    def __init__(self, attr_flags=0, class_flags=0):
+        self.attr_flags = attr_flags
+        self.class_flags = class_flags
+
+    def __call__(self, data):
+        """
+        Removes unused features or classes from the given data. Returns a new
+        data table.
+
+        Parameters
+        ----------
+        data : Orange.data.Table
+            A data table to remove features or classes from.
+
+        Returns
+        -------
+        data : Orange.data.Table
+            New data table.
+        """
+        if data is None:
+            return None, [None, None]
+
+        domain = data.domain
+        attrs_state = [purge_var_M(var, data, self.attr_flags)
+                       for var in domain.attributes]
+        class_state = [purge_var_M(var, data, self.class_flags)
+                       for var in domain.class_vars]
+
+        att_vars, att_res = self.get_vars_and_results(attrs_state)
+        cls_vars, cls_res = self.get_vars_and_results(class_state)
+
+        domain = Domain(att_vars, cls_vars, domain.metas)
+        return data.from_table(domain, data), [att_res, cls_res]
```

Comment:
Preprocessors need to return a new data table (e.g. `data2 = Preprocessor(data)`)
so they can be chained etc.

If the results dict is really needed, we can save the (last) results on the Remover instance:
```
remover = Remover(7, 7)
data2 = remover(data)
ar, cr = remover.attr_results, remover.cls_results
```

Another option is to add a parameter to `__call__(self, data, info=False)`
and only return the dicts when `info=True`